### PR TITLE
DAOS-3714: test  Delete entry from container ACL

### PR DESCRIPTION
**Pull request by amjustin13**
- Test that container delete command performs as expected with invalid inputs.
- Test that container delete command successfully removes principal in ACL.
- Test that container delete command fails with no permissions error (-1001) when trying to perform delete command as other user.

Signed-off-by: Justiniano-pagn <amanda.justiniano-pagn@intel.com>

### CODE REVIEW

File: src/tests/ftest/security/cont_delete_acl.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            except process.CmdError as err:
+                if "-1003" in err.result.stderr:
+                    self.log.info(
+                        "Found expected error %s with invalid principal %s",
+                        err.result.stderr,
+                        principal)
+                else:
+                    self.fail("delete-acl seems to have failed with \
+                        unexpected error: {}".format(err))
+        self.fail("container delete-acl command expected to fail.")
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_delete_acl.py:84]: Maybe I just don't understand this test well, but why do we have self.fail here? Will this test case always fail if it reaches the end of this test case? Same for test_no_user_permissions.

File: src/tests/ftest/security/cont_delete_acl.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                self.fail(
+                    "Found acl that was to be deleted in output: {}".format(
+                        self.daos_cmd.result.stdout))
+
+    def test_no_user_permissions(self):
+        """
+        JIRA ID: DAOS-3714
+
+        Test Description: Test that container delete command successfully
+            removes principal in ACL.
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_delete_acl.py:112]:
```suggestion
        Test Description: Test that container delete command doesn't
            remove principal in ACL without permission.
```
Along those lines...

**Review by shimizukko — APPROVED** [review]
Looks good to me.

File: src/tests/ftest/security/cont_delete_acl.py (hunk 1 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_delete_acl.py]: I re-implemented the error handling on my new patch.

File: src/tests/ftest/security/cont_delete_acl.py (hunk 2 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_delete_acl.py]: Fixed.

**Review by kjacque — DISMISSED** [review]

File: src/tests/ftest/security/cont_delete_acl.py [hunk 3]
```diff
@@ -52,6 +51,32 @@ def setUp(self):
         for entry in cont_acl:
             self.principals_table[entry.split(":")[2]] = entry
 
+    def error_handling(self, results, err_msg):
+        """Handle errors when test fails and when command unexpectedly passes.
+
+        Args:
+            results (CmdResult): object containing stdout, stderr and
+                exit status
+
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_delete_acl.py:60]: Description of err_msg is missing.

**Review by shimizukko — APPROVED** [review]
LGTM
Noticed that you used the same tag for all test cases.

File: src/tests/ftest/security/cont_delete_acl.py (hunk 3 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_delete_acl.py:60]: Ah.. thanks, I keep missing those details. I have another PR I'm working on, if I don't have to rebase anymore for this PR I'll add those changes in on that PR.

**Review by kjacque — APPROVED** [review]

**Comment by amjustin13** [comment]
Test Failures:
- Aggregation basic test failed on starting run_ior_with_pool and eventually timeout: https://jira.hpdd.intel.com/browse/DAOS-5502
- DaosServerTest: test_daos_server_reformat - Failing with DER_UNREACH(-1006): https://jira.hpdd.intel.com/browse/DAOS-5631
- osa_online_drain fails with timeout (1000s): https://jira.hpdd.intel.com/browse/DAOS-5538
- Error creating the /tmp/daos_dfuse/ dfuse mount point: https://jira.hpdd.intel.com/browse/DAOS-5559?filter=16118
- DAOS degrade mode test failed on rebuild: https://jira.hpdd.intel.com/browse/DAOS-5488

**Pull request merged**